### ISSUE TITLE
Use block to affect enumerations

### ISSUE THREAD

**Issue by greyblake** [albertosaurus/power_enum#12]
Instead of this:

``` ruby
Book.enumeration_model_updates_permitted = true
Book.create(:author => "Dostoyevsky F.M.", :name => "Crime and Punishment")
Book.create(:author => "Dostoyevsky F.M.", :name => "The Idiot" )
Book.enumeration_model_updates_permitted = false
```

It would be better to have syntax like this:

``` ruby
Book.affect_enumeration do
  Book.create(:author => "Dostoyevsky F.M.", :name => "Crime and Punishment")
  Book.create(:author => "Dostoyevsky F.M.", :name => "The Idiot" )
end
```

**Comment by baburdick** [comment]
+1

Here's Sergey's partial ad hoc implementation from another project:

``` ruby
  def affect_enumeration(model_class)
    model_class.enumeration_model_updates_permitted = true
    yield
  ensure
    model_class.enumeration_model_updates_permitted = false
  end
```

**Comment by albertosaurus** [comment]
Sounds like an outstanding idea.  Will do in the immediate future.

**Comment by albertosaurus** [comment]
Fixed in the 0.8.1 release.  See `update_enumerations_model`.
